core/execution/liquidation/engine.go

```diff
@@ -0,0 +1,307 @@
+	return defaultStrat.DeepClone()
+}
+
+// GetLegacyStrat is exported, same as defaul. This can be used for protocol upgrade
+// it most closely resebles the old behaviour (network attempts to close out fully, in one go)
+// this can be removed once protocol upgrade has completed.
+func GetLegacyStrat() *types.LiquidationStrategy {
+	return legacyStrat.DeepClone()
+}
+
+func New(cfg *types.LiquidationStrategy, mktID string, broker common.Broker, book Book, as common.AuctionState, tSvc common.TimeService, ml MarketLiquidity) *Engine {
+	// NOTE: This can be removed after protocol upgrade
+	if cfg == nil {
+		cfg = legacyStrat.DeepClone()
+	}
+	return &Engine{
+		cfg:    cfg,
+		broker: broker,
+		mID:    mktID,
+		book:   book,
+		as:     as,
+		tSvc:   tSvc,
+		ml:     ml,
+		pos:    &Pos{},
+	}
+}
+
+func (e *Engine) Update(cfg *types.LiquidationStrategy) {
+	if !e.nextStep.IsZero() {
+		since := e.nextStep.Add(-e.cfg.DisposalTimeStep) // work out when the network position was last updated
+		e.nextStep = since.Add(cfg.DisposalTimeStep)
+	}
+	// now update the config
+	e.cfg = cfg
+}
+
+func (e *Engine) OnTick(ctx context.Context, now time.Time) (*types.Order, error) {
+	if e.pos.open == 0 || e.as.InAuction() || e.nextStep.After(now) {
+		return nil, nil
+	}
+	minP, maxP, err := e.ml.ValidOrdersPriceRange()
+	if err != nil {
+		return nil, err
+	}
+	vol := e.pos.open
+	bookSide := types.SideBuy
+	side := types.SideSell
+	bound := minP
+	price := minP
+	if vol < 0 {
+		vol *= -1
+		side, bookSide = bookSide, side
+		price, bound = maxP, maxP
+	}
+	size := uint64(vol)
+	if size > e.cfg.FullDisposalSize {
+		// absolute size of network position * disposal fraction -> rounded
+		size = uint64(num.DecimalFromFloat(float64(size)).Mul(e.cfg.DisposalFraction).Round(0).IntPart())
+	}
+	available := e.book.GetVolumeAtPrice(bound, bookSide)
+	if available == 0 {
+		return nil, nil
+	}
+	// round up, avoid a value like 0.1 to be floored, favour closing out a position of 1 at least
+	maxCons := uint64(num.DecimalFromFloat(float64(available)).Mul(e.cfg.MaxFractionConsumed).Ceil().IntPart())
+	if maxCons < size {
+		size = maxCons
+	}
+	// get the block hash
+	_, blockHash := vegacontext.TraceIDFromContext(ctx)
+	idgen := idgeneration.New(blockHash + crypto.HashStrToHex("networkLS"+e.mID))
+	// set time for next order, if the position ends up closed out, then that's fine
+	// we'll remove this time when the position is updated
+	if size == 0 {
+		return nil, nil
+	}
+	e.nextStep = now.Add(e.cfg.DisposalTimeStep)
+	// place order using size
+	return &types.Order{
+		ID:          idgen.NextID(),
+		MarketID:    e.mID,
+		Party:       types.NetworkParty,
+		Side:        side,
+		Price:       price,
+		Size:        size,
+		Remaining:   size,
+		TimeInForce: types.OrderTimeInForceIOC,
+		Type:        types.OrderTypeLimit,
+		CreatedAt:   now.UnixNano(),
+		Status:      types.OrderStatusActive,
+		Reference:   "LS", // Liquidity sourcing
```

Comment:
Minor: given this will be visible in the data-node we sould just give it a reference thats less cryptic like "liquidity-sourcing".

Reply:
I think I copied this from the existing code, comment and all, so I was reluctant to change it, given that this reference is already in use for position resolution